src/LLVM.jl

```diff
@@ -7,79 +7,23 @@ using Libdl
 
 ## discovery
 
-using Libdl
-
-VERSION >= v"0.7.0-DEV.2576" || error("This version of LLVM.jl requires Julia 0.7")
-
-let
-    # find LLVM library
-
-    libllvm_paths = filter(Libdl.dllist()) do lib
-        occursin(r"LLVM\b", basename(lib))
-    end
-    if isempty(libllvm_paths)
-        error("""
-            Cannot find the LLVM library loaded by Julia.
-            Please use a version of Julia that has been built with USE_LLVM_SHLIB=1 (like the official binaries).
-            If you are, please file an issue and attach the output of `Libdl.dllist()`.""")
-    end
-    if length(libllvm_paths) > 1
-        error("""
-            Multiple LLVM libraries loaded by Julia.
-            Please file an issue and attach the output of `Libdl.dllist()`.""")
+export version
+
+# make sure we precompile again when LLVM changes (some definitions are version-dependent)
+global const libllvm = Sys.iswindows() ? :LLVM : :libLLVM
+Base.include_dependency(Libdl.dlpath(libllvm))
+
+const libllvm_version = Ref{VersionNumber}()
+function version()
+    if !isassigned(libllvm_version)
+        # FIXME: add a proper C API to LLVM
+        version_str = unsafe_string(
+            ccall((:_ZN4llvm16LTOCodeGenerator16getVersionStringEv, libllvm), Cstring, ()))
```

Comment:
xD

Reply:
Surprised it works on all platforms straight away!